tck/src/test/java/org/reactivestreams/tck/IdentityProcessorVerificationTest.java

```diff
@@ -115,6 +127,145 @@ public void required_spec104_mustCallOnErrorOnAllItsSubscribersIfItEncountersANo
+            int subscriberCount = subs.size();
+            switch (subscriberCount) {
+              case 0:
+                s.onSubscribe(createSubscription(awaitLatch, s, demand1));
+                break;
+              case 1:
+                s.onSubscribe(createSubscription(awaitLatch, s, demand2));
+                break;
+              default:
+                throw new RuntimeException(String.format("This for-test-purposes-processor supports only 2 subscribers, yet got %s!", subscriberCount));
+            }
+          }
+
+          @Override
+          public void onSubscribe(Subscription s) {
+            this.upstreamSubscription = s;
+          }
+
+          @Override
+          public void onNext(Integer elem) {
+            for (Subscriber<? super Integer> subscriber : subscribers) {
+              try {
+                subscriber.onNext(elem);
+              } catch (Throwable t) {
+                env.flop(t, String.format("Calling onNext on [%s] should not throw! See https://github.com/reactive-streams/reactive-streams-jvm#2.13", subscriber));
+              }
+            }
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            for (Subscriber<? super Integer> subscriber : subscribers) {
+              try {
+                subscriber.onError(t);
+              } catch (Exception ex) {
+                env.flop(ex, String.format("Calling onError on [%s] should not throw! See https://github.com/reactive-streams/reactive-streams-jvm#2.13", subscriber));
+              }
+            }
+          }
+
+          @Override
+          public void onComplete() {
+            for (Subscriber<? super Integer> subscriber : subscribers) {
+              try {
+                subscriber.onComplete();
+              } catch (Exception ex) {
+                env.flop(ex, String.format("Calling onComplete on [%s] should not throw! See https://github.com/reactive-streams/reactive-streams-jvm#2.13", subscriber));
+              }
+            }
+          }
+
+          private Subscription createSubscription(CountDownLatch awaitLatch, final Subscriber<? super Integer> s, final AtomicLong demand) {
+            final MySubscription sub = new MySubscription(awaitLatch, s, demand);
+            subs.add(sub);
+            subscribers.add(s);
+            return sub;
+          }
+
+          final class MySubscription implements Subscription {
+            private final CountDownLatch awaitLatch;
+            private final Subscriber<? super Integer> s;
+            private final AtomicLong demand;
+
+            public MySubscription(CountDownLatch awaitTwoLatch, Subscriber<? super Integer> s, AtomicLong demand) {
+              this.awaitLatch = awaitTwoLatch;
+              this.s = s;
+              this.demand = demand;
+            }
+
+            @Override
+            public void request(final long n) {
+              ex.execute(new Runnable() {
```

Comment:
I've been trying to incorporate the TCK into RxJava 2.0 but I could only test a couple of our components. The problem with the TCK I see is that it expects a certain shape of Publishers and Processors whereas the spec, in my opinion, allows much more variety. 

One indication to this that we have a `just` `Publisher` which only returns a single value. Testing it with the TCK passes half the tests and ignores the other because `just` will never emit an `onError`. The opposite, `error` `Publisher` will never emit a value but only an `onError` so I'm not certain the TCK test would skip the first half of it.

In addition, there is also the question if a `Processor` should do fine-grained request coordination or not. We have a controversial implementation in RxJava 2.0 where the processor requests Long.MAX_VALUE, tracks the request amount of its Subscribers and emits ˙`onError` to a `Subscriber` that can't keep up. This kind of `Processor` is untestable with the TCK as, even if the `Subscriber` it uses can keep up, expects the `Processor` to replay values and not just drop them if there are no `Subscriber`s.

I'm also guessing that having RxJava 2.0 as the dependency to support the TCK test is out of question.